PowerShellTools.HostService/ServiceManagement/Debugging/PowerShellDebuggingService.cs

```diff
@@ -762,6 +931,21 @@ public IEnumerable<CallStack> GetCallStack()
                             (string)psFrame.FunctionName.ToString(),
                             (int)psFrame.ScriptLineNumber));
                 }
+                else if(GetDebugScenario() == DebugScenario.Local_Attach)
+                {
+                    // local process attach debugging
+                    string currentCall = psobj.ToString();
+                    Match match = validStackLine.Match(currentCall);
+                    if (match.Success)
+                    {
+                        String funcall = match.Groups[1].Value;
+                        String script = match.Groups[3].Value;
+                        int lineNum = int.Parse(match.Groups[4].Value);
+
+                        callStackFrames.Add(new CallStack(script, funcall, lineNum));
+                    }
+                }
+

```

Comment:
Let me try to do the refactor to what Paul suggested and we may not even need to have multiple cases. Otherwise, I don't see why we couldn't have an assert here. I don't think I have a decent opinion on the matter so I'll let you take the ball on this one.